### PR TITLE
[devel_transport] Remove test_env

### PR DESCRIPTION
**Pull request by PrzemekWirkus**
Changes:
* Remove ```test_env.cpp``` completely.
* Add warning to ```test_env.h``` pointing to ```greentea-client/test_env.h``` replacement.
* Change ```greentea-client``` dependency to ```~0.1.4``` from ```^0.1.4```.

### CODE REVIEW

**Comment by PrzemekWirkus** [comment]
@0xc0170 @bogdanm Guys, please review. We are removing test_env from mbed-drivers forever :)

Destroy me!

File: module.json [hunk 1]
```diff
@@ -36,11 +36,11 @@
     "minar": "^1.0.0",
     "core-util": "^1.0.0",
     "compiler-polyfill": "^1.1.0",
-    "greentea-client": "^0.1.4"
+    "greentea-client": "~0.1.4"
```
Comment (bogdanm) [inline module.json]: If you list `greentea-client` as a regular dependency, it'll always be installed, so there's no need to also list it as a test dependency. I _think_ it should be a test dependency, but I haven't check if this is correct.

**Comment by bogdanm** [comment]
`greentea-client` is not backwards compatible with `test_env`, right? If so, removing `test_env` from `mbed-drivers` right now (before the release) is not a good idea, since that's bound to break things. I'd keep `test_env` in place for now, but also keep the warning you added to `test_env.h` in place, so that people know what to expect in the future.

**Comment by 0xc0170** [comment]
> greentea-client is not backwards compatible with test_env, right? If so, removing test_env from mbed-drivers right now (before the release) is not a good idea, since that's bound to break things. I'd keep test_env in place for now, but also keep the warning you added to test_env.h in place, so that people know what to expect in the future. 

It would be nice if we could make it backward compatible. In the worst case (not backward compatible, no integration for the upcoming release with this breaking changes), we could add a warning in test_env.h (assuming greentea client is released publicly)?

**Comment by PrzemekWirkus** [comment]
@bogdanm Bogdan, all devel_transport branches will be merged to master when you guys are ready.
We have few outstanding devel_transport branches for test tools.

So we will merge the same time everything.

**Comment by bogdanm** [comment]
Sorry, I haven't noticed initially that this is a PR against `devel_transport` instead of `master`. This is perfectly fine then, +1.

**Pull request merged**